src/plugins/displayLayout/CustomStringFormatter.js

```diff
@@ -0,0 +1,39 @@
+import printj from 'printj';
+
+export default class CustomStringFormatter {
+    constructor(openmct, itemFormat = null) {
+        this.openmct = openmct;
+
+        this.itemFormat = itemFormat;
+    }
+
+    format(datum, valueFormatter, valueMetadata) {
+        if (!this.itemFormat) {
+            return valueFormatter && valueFormatter.format(datum);
+        }
+
+        if (!this.itemFormat.startsWith('&')) {
+            return printj.sprintf(this.itemFormat, datum[valueMetadata.key]);
+        }
+
+        try {
+            const formatMap = this.openmct.telemetry.formatService.formatMap;
```

Comment:
`formatService.formatMap` is an internal detail we shouldn't expose here. It depends on legacy API that is going to be removed.

Let's add a new API function - `openmct.telemetry.getFormatter(key)`.

Reply:
Done